GOB: add option to choose savepath for variables.dmp

BJNFNE opened the pull request:
This is allows the user either with just dumpVars command to save in the ScummVM directory, or if the user want to dump the variables into an specific folder.

- Benefit:
* If the variables dump will be saved in an other folder then Git/GitHub Desktop won't show it to upload it to the repository.
* Also for compare variable dumps between multiple versions, then save on in this directory and one in this directory for comparing them.

lephilousophe commented:
Do you have any plans on using this?
Which feature do you plan to do with this?

BJNFNE commented:
I am so far done with this PR. The main usage case i have added it for GitHub/Repository to avoid to adding it by accident, see here #5845 i have fixed it in this PR in the code instead of adding it to .gitignore.

For the Dailybuild's it don't have to use the choosen savepath for variables.dmp, for these builds dumpVars itself is enough.

the 2nd idea for the comparing, so that we can have 2 or more variables.dmp of various versions of an game and save their variables.dmp at different paths.

Also i thought about an readVars() function which reads in the variables.dmp in GobConsole directly and print out the contents / variables in the Console instead of manually opening the variables dump in an Editor or look at it with strings, but i have no idea yet how to do it.

sev- commented:
We do not need a code like this. It is dead, unused, but still needs to be maintained.

Please do not add useless features. Only the engine developers may want/need to expand the debugging tools.

If you have a real-world usage for this, e.g., use for adding support of a specific game of a variant, then we will gladly accept the new code.

You did a good work of putting the detection tables in order, and the work there is mostly complete unless we find a new variant.

Please do not pull the team resources on things that have no real application.
